src/java/com/google/cloud/bigquery/dwhassessment/hooks/logger/EventLogger.java

```diff
@@ -0,0 +1,241 @@
+    if (baseDir == null) {
+      return null;
+    }
+
+    try {
+      return new DatePartitionedLogger(new Path(baseDir), conf, QUERY_EVENT_SCHEMA, clock);
+    } catch (IOException e) {
+      LOG.error("Unable to initialize logger, logging disabled.", e);
+    }
+
+    return null;
+  }
+
+  private String constructFileName() {
+    return "dwhassessment_" + loggerId + "-" + logFileCount + ".avro";
+  }
+
+  private void handleTick() {
+    try {
+      maybeRolloverWriterForDay();
+    } catch (IOException e) {
+      LOG.error("Got IOException while trying to rollover: ", e);
+    }
+  }
+
+  private void maybeRolloverWriterForDay() throws IOException {
+    if (writer == null || !logger.getNow().equals(writerDate)) {
+      if (writer != null) {
+        // Day changes over case, reset the logFileCount.
+        logFileCount = 0;
+        IOUtils.closeQuietly(writer);
+        writer = null;
+      }
+      // increment log file count, if creating a new writer.
+      ++logFileCount;
+      writer = logger.getWriter(constructFileName());
+      writerDate = DatePartitionedLogger.getDateFromDir(writer.getPath().getParent().getName());
+    }
+  }
+
+  private void writeEventWithRetries(GenericRecord event) {
+    for (int retryCount = 0; retryCount <= MAX_RETRIES; ++retryCount) {
+      try {
+        maybeRolloverWriterForDay();
+        writer.writeMessage(event);
+        writer.hflush();
+        return;
+      } catch (IOException e) {
+        // Something wrong with writer – close and reopen.
+        IOUtils.closeQuietly(writer);
+        writer = null;
+        reportRetryState(event, retryCount, e);
+        waitForRetry(retryCount);
+      }
+    }
+  }
+
+  private void reportRetryState(GenericRecord event, int retryCount, IOException writeException) {
+    if (retryCount < MAX_RETRIES) {
+      LOG.warn(
+          "Error writing proto message for query {}, eventType: {}, retryCount: {},"
+              + " error: {} ",
+          event.get("QueryId"),
+          event.get("EventType"),
+          retryCount,
+          writeException.getMessage(),
+          writeException);
+      LOG.trace("Exception", writeException);
+    } else {
+      LOG.error(
+          "Error writing proto message for query {}, eventType: {}: ",
+          event.get("QueryId"),
+          event.get("EventType"),
+          writeException);
+    }
+  }
+
+  private void waitForRetry(int retryCount) {
+    try {
+      // 0 seconds, for first retry assuming fs object was closed and open will fix it.
+      Thread.sleep(1000L * retryCount * retryCount);
+    } catch (InterruptedException e1) {
+      LOG.warn("Got interrupted in retry sleep.", e1);
+    }
+  }
+
+
+  public void shutdown() {
+    if (logWriter != null) {
+      logWriter.shutdown();
+      try {
+        logWriter.awaitTermination(SHUTDOWN_WAIT_TIME, TimeUnit.SECONDS);
+      } catch (InterruptedException e) {
```

Comment:
The simplest way to handle `InterruptedException` is to call `Thread.currentThread().interrupt()` before `LOG.warn`.

Reply:
Done